src/bin/cql-stress-cassandra-stress/java_generate/distribution/mod.rs

```diff
@@ -9,6 +18,33 @@ pub trait Distribution: Send + Sync {
     fn set_seed(&self, seed: i64);
 }
 
+/// A thread_local wrapper for [java_random::Random].
+/// Used by distributions to implement `atomic` sampling.
+struct ThreadLocalRandom {
+    rng: ThreadLocal<RefCell<Random>>,
+}
+
+impl ThreadLocalRandom {
+    fn new() -> Self {
+        Self {
+            rng: ThreadLocal::new(),
+        }
+    }
+
+    fn get(&self) -> RefMut<'_, Random> {
+        self.rng
+            .get_or(|| {
+                RefCell::new(Random::with_seed(
+                    SystemTime::now()
+                        .duration_since(UNIX_EPOCH)
+                        .map(|duration| duration.as_millis() as u64)
+                        .unwrap_or_default(),
+                ))
```

Comment:
c-s generates deterministic values only when partition_key generator seeds are sampled from DETERMINISTIC distribution (i.e. `SeqDistribution`). This approach is usually taken when populating the database with write routine.

For example:
```
./cassandra-stress write n=100 -pop dist=SEQ(1..100)
```
will generate the exact same set of 100 rows each time it's called.

Now, the data can be validated with `read` routine, for example:
```
./cassandra-stress read n=100 -pop dist=SEQ(1..100)
./cassandra-stress read n=100 -pop dist=UNIFORM(1..100)
./cassandra-stress read n=100 -pop dist=GAUSSIAN(30..70)
```
All of these workloads will successfully validate the data (inserted to DB with the write workload mentioned above).
To be more specific: any read workload that samples pk seeds from the subset of (1..100) range will successfully validate the data.

However, note that executing the following comments may not (and in most of the cases will not) be successful:
```
./cassandra-stress write n=100 -pop dist=UNIFORM(1..100)
./cassandra-stress read n=100 -pop dist=UNIFORM(1..100)
```
It's because the partition_key seed distribution is not deterministic. But as I mentioned, it is almost always the case that we populate the DB using some deterministic distribution.

When it comes to `Generator`s (which use `Distribution`s under the hood) - they are explicitly  seeded during row generation. See the row generation logic introduced in the following PR: https://github.com/scylladb/cql-stress/pull/45/commits/866a29c1ae496cfba9b8689a120ebef4caef0d21#diff-65a79d7c79bfc82d1f21b385ad5a7d03c5b12a1fcbc03113e41ea2c14ccbe544R27. `RowGeneratror` samples the `pk_seed` from the distribution shared by all of the tasks/threads (the distribution specified by the user with `-pop dist=...` option). The row values are fully dependent on the `pk_seed` sampled from this distribution.

Answering your question - yes, such seeding will stay in tune with c-s seeds.